x-pack/plugins/observability/public/pages/landing/landing.tsx

```diff
@@ -4,29 +4,35 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
-import React from 'react';
+import { DISCOVER_APP_ID } from '@kbn/deeplinks-analytics';
+import React, { useEffect } from 'react';
 import { useHasData } from '../../hooks/use_has_data';
 import { useKibana } from '../../utils/kibana_react';
 
 export function LandingPage() {
   const { hasDataMap, isAllRequestsComplete } = useHasData();
   const {
-    application: { navigateToUrl },
+    application: { navigateToUrl, navigateToApp },
     http: { basePath },
   } = useKibana().services;
 
-  if (isAllRequestsComplete) {
-    const { apm, infra_logs: logs } = hasDataMap;
-    const hasApmData = apm?.hasData;
-    const hasLogsData = logs?.hasData;
+  useEffect(() => {
+    if (isAllRequestsComplete) {
+      const { apm, infra_logs: logs } = hasDataMap;
+      const hasApmData = apm?.hasData;
+      const hasLogsData = logs?.hasData;
 
-    if (hasLogsData) {
-      navigateToUrl(basePath.prepend('/app/discover'));
-    } else if (hasApmData) {
-      navigateToUrl(basePath.prepend('/app/apm/services'));
-    } else {
-      navigateToUrl(basePath.prepend('/app/observabilityOnboarding'));
+      if (hasLogsData) {
+        navigateToApp(DISCOVER_APP_ID, {
+          deepLinkId: 'log-explorer',
```

Comment:
I see why it works, `navigateToApp` internally retrieves the app deepLinks, flattens them and matches by the deepLinkId, that's why it doesn't need to get the full deepLinkId `discover:log-explorer`. It works correctly as you implemented it!